### DISCUSSION TITLE
Feature: ability to make 1 or more streams on montage suddenly larger than others on high motion

### DISCUSSION THREAD

**Discussion by connortechnology** [ZoneMinder/zoneminder#4161] [Ideas]
I've roughed in current score into the stream data. . 

Can we tell gridstack that suddenly these ones are larger? (maybe double).

Or do we maintain a separate grid stack at the top, which is always empty, until a stream is moved to it?

brainstorming.

**Comment by Aleksanderk123** [comment]
I did some very basic testing, and it looks like this can be managed quite easily with some simple CSS.
Wouldn't this, although beeing just a super simple test, do what you want?

```
.monitor:has(div.alarm) {
  border: 3px solid red !important;
  position: absolute;
  top: 14rem !important;
  background: #000;
  display: inline-table;
  left: 25% !important;
  width: 50% !important;
}
```

**Comment by IgorA100** [comment]
@connortechnology 
I don't quite understand your idea :(

**Comment by connortechnology** [comment]
This is a request from a client.  So normally they have a big tv on the wall with montage watching all the streams. And these are schools, and there is a panic button in each classroom.  WHen someone hits that button, the cameras relating to that room are supposed to get bigger than the others to take over more real estate on the screen.  We already have the red border on alarm, this is kinda next level from that.

**Reply by IgorA100** [reply]
> WHen someone hits that button, the cameras relating to that room are supposed to get bigger than the others to take over more real estate on the screen.

The idea is interesting!
But how do you know where to place the enlarged image?
For example, you need to enlarge an image located in the lower left corner and it is completely out of the screen's visibility? What should you do? It will definitely have to be moved somewhere....
And what if two or three or more images are enlarged?
How many times should the image be enlarged?
And one more thing. Let's say one image has a conditional size of 4x3, the second 8x6 (i.e. it is already configured to be displayed as enlarged), the third 12x9 and it already takes up 2/3 of the screen. You can enlarge the first 4x3 to 8x6, but then the third 12x9 image will have to be reduced or hidden somewhere... What if a signal to enlarge the third image is received next, then there will be a complete collapse.....
So many questions...

**Reply by IgorA100** [reply]
So far I have one thought:
It is necessary to create a new Layout in Auto mode on which to place cameras with an alarm. After the alarm ends, delete the alarm Layout and display the previous (which was before the alarm) Layout

**Comment by connortechnology** [comment]
Yeah it's not trivial.  I was thinking a secondary grid, above the normal one.  Since it would normally be empty, nothing would show.  As cameras are triggered they would get added.  It would be up to the grid to try to show as many as possible filling the screen.  But in the event of an actual emergency I think this would end up in chaos.  I think we would have to do some experiments.

**Comment by IgorA100** [comment]
> But in the event of an actual emergency I think this would end up in chaos.

There shouldn't be global chaos, but what if 40 cameras go off? :))) They won't all fit on the monitor. And if they do, everything will slow down terribly.....
And you just need to set a delay before displaying the cameras on the secondary grid,. For example, collect signals in a buffer and once every X sec. draw on the grid.
